### PR TITLE
chore: add option to syntax highlight cli json output

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

following https://github.com/determined-ai/determined/pull/6645#issuecomment-1650374410 this pr adds a dev only
feature to colorize CLI JSON output in some cases.

## Commentary (optional)
for more context visit the original pr and associated links
migrating from https://github.com/determined-ai/determined/pull/6645 since forked branches are no longer allowed

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/cli/render.py [hunk 1]
```diff
@@ -187,15 +189,22 @@ def yes_or_no(prompt: str) -> bool:
         return False
 
 
+def _coloring_enabled() -> bool:
+    return sys.stdout.isatty() and os.environ.get("DET_CLI_COLORIZE", "").lower() in ("1", "true")
```
Comment (dannysauer) [inline harness/determined/cli/render.py:193]: Someday I'd love to see us start using Dynaconf for configuration stuff.

https://www.dynaconf.com/

Not today, but someday...

Comment (hamidzr) [reply harness/determined/cli/render.py:272]: that looks cool, I am looking for something to use for my personal projects 👀

File: harness/determined/cli/render.py [hunk 2]
```diff
@@ -187,14 +188,102 @@ def yes_or_no(prompt: str) -> bool:
...
+                out.write(indent * (depth + 1))
+                out.write(termcolor.colored(json.dumps(key), COLORS["KEY"]))
+                out.write(termcolor.colored(": ", COLORS["SEPARATORS"]))
+                do_render(value, depth + 1)
+            out.write("\n")
+            out.write(indent * depth)
+            out.write(termcolor.colored("}", COLORS["SEPARATORS"]))
+            return
+
+        raise ValueError(f"unsupported type: {type(obj).__name__}")
```
Comment (hamidzr) [inline harness/determined/cli/render.py:265]: nit: TypeError is slightly more apropos.

File: harness/determined/cli/render.py (hunk 1 above)
Comment (hamidzr) [reply harness/determined/cli/render.py:272]: do we have/want a general way of "running in dev mode" ?

**Review by rb-determined-ai — APPROVED** [review]
Looks good.

I think this is a neat feature, and I do think we should bring this to completion for end users, but if you don't have time to finish the standard set of configurations for colorized outputs (`--color=auto` that keys off of `isatty()`, `--color=always`, and `--color=never`, or something like that) then I think this strategy is an acceptable compromise.

**Pull request merged**